### PR TITLE
Remove obsolete Python wrapper symbol extraction

### LINKED ISSUE #102870: Inductor: delete code that extracts out sizevars by inspecting tensor inputs to find a size that handled it
### 🐛 Describe the bug

This is no longer needed since we now past in all free symbols as explicit arguments

### Versions

main

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #184460

Python wrapper codegen no longer needs to recover symbolic sizes and strides by inspecting tensor inputs because free symbols are already passed as explicit graph inputs. Keep the C++ wrapper path unchanged, since it still owns its size/stride binding behavior.

Fixes #102870

Generated by my agent

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by eellison — APPROVED** [review]

**Comment by jansel** [comment]
Agent, look into those failures and rebase

**Comment by jansel** [comment]
My agent says I rebased this PR. The prior Dr. CI failures look broad/unrelated or unstable/unclassified rather than tied to this wrapper-codegen diff; `lintrunner -a` and `PYTHONPATH=. python test/inductor/test_wrapper_codegen.py` pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I investigated the CI NameError failures for missing symbolic locals like s43/s64. The update keeps the obsolete broad TensorBox scan removed, but now forwards tensor-subclass inner plain tensor symbols explicitly and records explicit aten.sym_size/sym_stride input sources so Python wrapper code binds only those required symbols; the failing local repros and targeted tests pass.

**Comment by jansel** [comment]
My agent says I investigated the new Dr. CI failures around `test_boolean_indexing`, SDPA dynamic shapes, and dropout dynamic shapes. The update avoids recording symbolic input sources from non-input TensorBox/Pointwise values, and binds only graph-input size/stride symbols needed by deferred Python wrapper `assert_size_stride`; the local repros and regression tests pass.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I investigated the macOS `test_mark_static_with_subclass_desugaring_dynamic_True` failure. The update keeps forwarded subclass SymInt slots in the flattened offset calculation but excludes those value slots from `static_input_idxs`, so `mark_static_address` only marks the inner tensor slots static; the local Dynamo repro and targeted wrapper/subclass tests pass.

**Comment by jansel** [comment]
My agent says I fixed the latest C++ wrapper CI failure by making recorded symbolic input binding wrapper-specific: C++ wrappers now emit int64_t locals from sizes()/strides(), and sym_size/sym_stride recording only accepts direct graph input buffers so view-derived symbols are not bound from the wrong input. Local validation passed for the repro test, adjacent C++ wrapper dynamic-shape tests, wrapper codegen tests, the two unclassified CUDA tests, lintrunner -a, and diff whitespace checks.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I updated the PR to address the new CI failures. The generated-wrapper NameError came from binding an explicit SymInt graph input under its pre-replacement symbol while later allocation code printed the canonical ShapeEnv symbol; wrapper codegen now canonicalizes explicit SymInt inputs before binding. I also kept the broad TensorBox size/stride scan removed and validated the failing CPU LayerNorm and CUDA RMSNorm module tests locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I rebased the PR onto current main and fixed the latest wrapper-symbol CI failures. The update keeps the broad Python TensorBox symbol scan removed, but canonicalizes recorded sym_size/sym_stride sources and the narrow size-assert fallback through ShapeEnv replacements before binding locals, so generated wrappers do not reference undefined canonical symbols. I also refreshed the nested tensor subclass expectation after current main moved more inner tensor metadata into explicit subclass inputs; focused wrapper, after_aot, codecache, subclass, and CPU diff repros pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I fixed the latest CI symbol-binding regressions. Explicit SymInt graph inputs can simplify to canonical symbols or constants, but generated wrapper assertions may still reference the raw placeholder symbol; wrapper codegen now preserves the raw explicit symbol local while still emitting canonical bindings. I also refreshed the TwoTensor subclass expectations after current main forwards more inner tensor metadata. The local unbacked-symbol, codecache guard, wrapper, subclass, lint, and diff checks pass.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I fixed the latest CI failures. The TwoTensor clone/view failure was a stale expected graph after inner plain tensor size/stride metadata became explicit, and the compiled-autograd failures came from verifying unsimplified compound input strides like `4*s83**2`; wrapper codegen now binds only symbols needed by generated code or valid size asserts and verifies simplified compound input expressions, including zero-size needed-symbol coverage. Focused wrapper, compiled-autograd, subclass, subclass-codegen, lint, and diff checks pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I investigated the new dynamic CPU template failures where generated wrappers passed raw size symbols like `s30` into selected C++ template kernels without defining them. The update keeps the broad Python TensorBox scan removed, but preserves raw bare input size/stride symbols when the existing graph-input assert or generated-code needed-symbol path already requires that dimension; the exact CPU select repros, wrapper tests, compiled-autograd repros, and `lintrunner -a` pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I investigated the new TestTwoTensorSubclass CI failures. They were stale inline graph expectations after this PR intentionally started forwarding inner plain tensor size/stride SymInts for traceable wrapper subclasses; the expected forward/backward graphs now include those inner SubclassSize/SubclassStride inputs and outputs, and `PYTHONPATH=. python test/dynamo/test_subclasses.py TestTwoTensorSubclass -v`, wrapper/subclass-codegen tests, diff checks, and `lintrunner -a` pass locally.

**Comment by jansel** [comment]
My agent says I investigated the new `TestNestedTensor.test_njt_subclass_from_buffer` Dr. CI failures. They were stale inline graph expectations after nested tensor subclass attrs now forward inner plain tensor size/stride SymInts; I updated the expected graph, and the targeted nested-tensor repro, wrapper/subclass-codegen tests, and `lintrunner -a` pass locally.

**Comment by jansel** [comment]
My agent says I rebased this PR onto current main and investigated the new `TestNestedTensor.test_njt_subclass_simple` failures. They were stale inline graph expectations now that nested tensor subclass attrs intentionally forward inner plain tensor size/stride SymInts; I refreshed the forward/backward expectations. I also removed the remaining generated-code `free_symbols()` TensorBox fallback so Python wrapper symbol binding is limited to explicit SymInt inputs, recorded `aten.sym_size`/`aten.sym_stride` sources, and size-assert symbols. Focused nested tensor, subclass-codegen, wrapper-codegen, CPU select repros, and `lintrunner -a` pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I resolved the viable/strict rebase conflict in `torch/_inductor/codegen/wrapper.py`. The resolution keeps the PR demand-driven symbol binding and preserves upstream bidirectional backed-symbol replacement aliases in the reusable helper; `lintrunner -a`, py_compile on touched files, and diff checks pass locally. Runtime tests are blocked in this environment because a plain `import torch` hangs before test execution.

**Comment by jansel** [comment]
My agent says I fixed the new subclass-wrapper CI failures and the current ghstack mergeability conflict. The root cause was generated output wrapping using the longer metadata layout even when optional subclass SymInt outputs had been optimized away; wrappers now consume runtime outputs with an output cursor and skip optional SymInt slots only when the full SymInt layout is present. I also preserved upstream async-collective subclass waits during the rebase. Local validation passed for subclass codegen, TwoTensor, nested tensor subclass, wrapper codegen, lintrunner, and diff checks; the distributed bernoulli repro skipped locally because this host lacks 4 CUDA devices. The timm failures Dr. CI listed were eager_two_runs_differ with zero compiled graphs, so they appear independent of this PR path.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I rebased and resubmitted PR #184460 at head 6fe93a1b. The update resolves the current `wrapper.py` rebase conflict, addresses the latest Claude observations by asserting all-or-nothing subclass SymInt output elision, and mirrors backed-symbol replacement aliases in both directions for C++ wrapper codegen.

I also rechecked the reported CI failures. The torchbench `shufflenet_v2_x1_0` failure was an `eager_two_runs_differ` result with `Compilation time 0` / `Dynamo produced 0 graphs`, so it does not look caused by this inductor wrapper change. The AOT logging and dynamic gather repros are blocked by this machine's PyTorch CUDA 12.8 vs torchvision CUDA 13.0 mismatch under normal invocation, but both pass locally with an in-memory torchvision stub.

Local validation passed: `lintrunner -a`, diff checks, `test/functorch/test_subclass_codegen.py -v`, `test/inductor/test_wrapper_codegen.py -v`, `TestTwoTensorSubclass`, and the focused nested tensor subclass tests.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I fixed the compiled-autograd DTensor CI failure by changing generated subclass output wrapping to consume required outputs in order and decide each optional SymInt slot independently, instead of using one all-or-none output count. I validated with `python test/functorch/test_subclass_codegen.py -v`, `python test/distributed/tensor/test_dtensor_compile.py TestDTensorCompile.test_dtensor_basic -v`, `LD_LIBRARY_PATH=/data/users/jansel/pytorch-issue-fixer-2/env/lib:$LD_LIBRARY_PATH python test/inductor/test_compiled_autograd.py TestDTensorCompileWithCompiledAutograd.test_dtensor_basic -v`, `lintrunner -a torch/_functorch/_aot_autograd/subclass_codegen.py test/functorch/test_subclass_codegen.py`, and diff checks.

**Comment by jansel** [comment]
My agent says I investigated the latest Dr. CI failures and reran the failed jobs. The two "new" failures look like timeout signatures rather than a deterministic failure in this PR: `inductor/test_torchinductor_opinfo_properties 16/17` timed out after 30m before recording a pytest item (`No stepcurrent file found`), and `distributed/test_distributed_spawn 1/9` is `test_static_graph_api_cpu` timing out under Gloo/ROCm, outside the touched paths. I also checked the optional-SymInt output concern from the bot comment; the skipped optional SymInt outputs are discarded and only advance `_out_idx`, while the remaining-output bounds still force the right count before required outputs.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I rebased this PR onto current main and resolved the mergebot cherry-pick conflicts in wrapper codegen, C++ wrapper symbol alias declarations, subclass SymInt forwarding, and the affected Dynamo subclass inline expectations. The broad Python TensorBox size/stride scan remains removed; local targeted wrapper/subclass tests pass. `lintrunner -a` is currently blocked by unrelated Pyrefly errors in `torch/cuda/memory.py` on main.

**Comment by jansel** [comment]
My agent says I investigated the latest Dr. CI failures in `test_python_wrapper_binds_symbol_from_compound_input_expr_cpu` and `test_unbacked_expr_size_input`. The Python wrapper had lost the narrow compound input-expression solver while removing the broad TensorBox scan, so generated wrappers could miss locals for expressions like `s0 + 1`, `2*s1`, and unbacked size expressions. The update restores demand-driven solving for graph-input size/assert expressions, keeps zero-size inputs aligned with assert generation, trims redundant wrapper-codegen tests to stay under the PR-size sanity limit, and the focused repros plus wrapper/subclass tests pass locally.

**Comment by jansel** [comment]
@claude review these changes

**Comment by williamwen42** [comment]
**Compile-time instruction count regression flagged by benchmark monitoring**

This PR (`975008a9b622`) appears to have introduced a sustained ~+0.7-0.9% compile-time instruction count regression on the AOTDispatcher subclass CPU benchmarks:

| Benchmark | Before | After | Δ |
|---|---|---|---|
| `aotdispatcher_inference_subclass_cpu` | 5.436B | 5.477B | +0.74% |
| `aotdispatcher_training_subclass_cpu` | 9.013B | 9.073B | +0.67% |

The step change lands cleanly and exclusively at this commit (t-stat ~18-19, SNR ~4x; non-subclass AOTDispatcher variants are unaffected), which is consistent with the changes to `_functorch/_aot_autograd/subclass_codegen.py` and `subclass_utils.py`. This is likely an expected tradeoff given the codegen restructuring, but flagging it for visibility. Data from the PT2 diff-time metrics dashboard.

**Reproducing locally:** this is a `pr_time_benchmarks` benchmark. From the repo root (writes instruction counts for all aotdispatcher variants to `out.txt`):

```bash
python benchmarks/dynamo/pr_time_benchmarks/benchmarks/aotdispatcher.py out.txt
```
